gcloud/pubsub/subscription.py

```diff
@@ -67,12 +75,16 @@ def from_api_repr(cls, resource, client, topics=None):
         if topics is None:
             topics = {}
         topic_path = resource['topic']
-        topic = topics.get(topic_path)
-        if topic is None:
-            # NOTE: This duplicates behavior from Topic.from_api_repr to avoid
-            #       an import cycle.
-            topic_name = topic_name_from_path(topic_path, client.project)
-            topic = topics[topic_path] = client.topic(topic_name)
+        if topic_path == cls._DELETED_TOPIC_PATH:
+            # Use a name which cannot conflict ('#' not allowed).
+            topic = client.topic(None)
```

Comment:
Unless we re-plumb `Subscription.__init__` to take the `project` and `client` directly, I don't see how, unless you think we should make the `topic` optional (*ick*):  in that case, `Subscription.__init__` would copy `project` and `client` from the `topic`, but only if passed, and `Subscription.from_api_repr` would set the private attrs directly.  I don't see a win vs the current state, cleanliness-wise.